csharp/src/Ice/IObjectPrx.cs

```diff
@@ -503,6 +513,16 @@ public void IceUpdateRequestHandler(IRequestHandler? previous, IRequestHandler?
             }
         }
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected static async Task IceAwaitResponse(Task<IncomingResponseFrame> task,
```

Comment:
The idea here is because readResponse just "wraps" a static method, the C# compiler uses a single constant delegate for each call (and does not allocate a new delegate each time)?

If yes, add a comment!

Reply:
Getting rid of these functions in to add back static local functions.